lfs_util.h

```diff
@@ -43,6 +43,9 @@ extern "C"
 {
 #endif
 
+#ifndef LFS_THREAD_SAFE
```

Comment:
A simple comment here such as "Enables thread-safe wrappers using the lock/unlock callbacks in lfs_config" could go a long way